### PR TITLE
build and build_releases VERSION override

### PR DESCRIPTION
**Pull request by ashcrow**
Currently a source checkout requires extra work to build the artifacts. Example:

```
[steve@qward ignition-0.23.0] $ ./build
Using version from git...
fatal: Not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

Overriding doesn't work out of the box:

```
[steve@qward ignition-0.23.0]$ VERSION=1 ./build
fatal: Not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

This change allows for ``VERSION`` to be overridden in both ``build`` and ``build_releases`` via:

```
[steve@qward ignition-0.23.0] $ VERSION=0.23.0 ./build
```

If ``VERSION`` is not set then ``VERSION`` defaults to what it is able to get via ``git`` (or fails as it did before if not a git checkout).

### CODE REVIEW

File: build [hunk 1]
```diff
@@ -5,10 +5,17 @@ set -eu
...
 GLDFLAGS=${GLDFLAGS:-}
+
+if [ -z ${VERSION+a} ]; then
+	echo "Using version from git..."
+	VERSION=$(git describe --dirty)
+fi
+
 GLDFLAGS+="-X github.com/coreos/ignition/internal/version.Raw=${VERSION}"
 
+
```
Comment (arithx) [inline build]: extra blank lines

**Comment by ashcrow** [comment]
Updated.

**Comment by ashcrow** [comment]
Did ``tailor`` hang?

**Comment by lucab** [comment]
@ashcrow it had a temporary hiccup yesterday. If you force-push this again, it will likely pick it up.

**Comment by ashcrow** [comment]
Thanks @lucab. I updated the commit message and force pushed.

**Pull request merged**